sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1249,6 +1249,15 @@ class Lasso(ElasticNet):
     To avoid unnecessary memory duplication the X argument of the fit method
     should be directly passed as a Fortran-contiguous numpy array.
 
+    Alpha must be a positive float for regularization strength.
```

Comment:
This sentence should be added in the `alpha` section instead I think. Now, we tend to provide the interval `[0, inf)` in this case.